sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainer.java

```diff
@@ -793,7 +795,7 @@ public void enableGlobalThroughputControlGroup(ThroughputControlGroupConfig grou
      *
      */
     public void openConnectionsAndInitCaches() {
-        blockVoidResponse(this.asyncContainer.openConnectionsAndInitCaches());
+        this.asyncContainer.openConnectionsAndInitCaches().subscribe();
```

Comment:
I am not particularly sure on how to block on a flux generated by a sink. Blocking here would cause an indefinite hang. Any ideas here?

Reply:
when using blockLast? do you ever signla complete on the sink?

Reply:
https://github.com/Azure/azure-sdk-for-java/blob/bbb09a6f10d4deee9ccc58fda29b1364de007c1e/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BulkExecutor.java#L862-L885